components/constellation/constellation.rs

```diff
@@ -2439,6 +2451,46 @@ where
         }
     }
 
+    #[cfg_attr(
+        feature = "tracing",
+        tracing::instrument(skip_all, fields(servo_profiling = true), level = "trace")
+    )]
+    fn handle_disentangle_messageports(&mut self, port1: MessagePortId, port2: MessagePortId) {
+        if let Some(info) = self.message_ports.get_mut(&port1) {
+            info.entangled_with = None;
+        } else {
+            warn!(
+                "Constellation asked to disentangle unknown messageport: {:?}",
+                port1
+            );
+        }
+        if let Some(info) = self.message_ports.get_mut(&port2) {
+            info.entangled_with = None;
+            match &mut info.state {
+                TransferState::Managed(router_id) |
+                TransferState::CompletionInProgress(router_id) => {
+                    // We try to disentangle the other port now,
+                    // and if it has been transfered out by the time the message is received,
+                    // it will be ignored,
+                    // and disentanglement will be completed as part of the transfer.
+                    if let Some(ipc_sender) = self.message_port_routers.get(router_id) {
+                        let _ = ipc_sender.send(MessagePortMsg::CompleteDisentanglement(port2));
+                    } else {
+                        warn!("No message-port sender for {:?}", router_id);
+                    }
+                },
+                _ => {
+                    // Note: the port is in transfer, disentanglement will complete along with it.
```

Comment:
I'd like to note that I've tried, and wasn't able to, hit this code path in a test.